### DISCUSSION TITLE
token doesn't store in cookies if `sameSite` set to `none`

### DISCUSSION THREAD

**Discussion by alanyong91** [nextauthjs/next-auth#2092] [Help]
## Description 🐜

I tried to change cookies and set `httpOnly` to `false`. 

```
export default NextAuth({
  providers: [
    Providers.Facebook({
      clientId: process.env.FACEBOOK_APP_ID,
      clientSecret: process.env.FACEBOOK_APP_SECRET
    }),
  ],
  database: process.env.MONGODB_CONNECTION,
  session: {
    jwt: true,
    maxAge: 24 * 60 * 60 // 1 day
  },
  jwt: {
    secret: process.env.JWT_SECRET
  },
  cookies: {
    sessionToken: {
      name: process.env.TOKEN_NAME!,
      options: {
        httpOnly: false,
        sameSite: 'none',
        secure: false
      }
    }
  }
})
```

I tried to login with Facebook but the token doesn't store in cookies. If I set `sameSite` to other value like `lax` then it will store the token in cookies.

**Comment by balazsorban44** [comment]
Could you explain your usecase? This seems like a bad idea.

**Comment by alanyong91** [comment]
Not sure what you meant by use case. Basically what I do is change cookies name and set `sameSite` value to `none` in `NextAuth` configuration. My code is in the description.

**Comment by balazsorban44** [comment]
I meant what's the reason for these changes? `httpOnly: false` is certainly not something you want to do for the session cookie.


From the few lines of code and explanation you provided, the cookie changes seem unnecessary at all. Remember that the cookie settings are for advanced usecases. What problem are you trying to solve here?

From the docs:
https://next-auth.js.org/configuration/options#cookies
> Using a custom cookie policy may introduce security flaws into your application and is intended as an option for advanced users who understand the implications. Using this option is not recommended

**Comment by alanyong91** [comment]
Oh! I was trying set `httpOnly` to `true`. I will change that in the description.

Anyway, I want change the cookies name in order to match with backend token, which I need to update `options` as well, otherwise it will throw typescript error. 

And the reason `httpOnly` has to be `false` is because I need to retrieve cookies from client side.

**Comment by balazsorban44** [comment]
We are not going to recommend httponly false. (it is set to true by default)

See:
https://owasp.org/www-community/HttpOnly

I am still not able to pinpoint your exact usecase, but I don't think this issue derives from a bug in our implementation. I don't understand why you would need to change the name to match your backend.

**Comment by bharatgupta99** [comment]
@balazsorban44  I want to use authentication locally(backend(localhost:8080) and frontend (localhost:3000) domains are different), that is my use case for setting `sameSite` to none.

**Reply by shamun-khatri** [reply]
@bharatgupta99 you have got any solution for this problem? I am also suffering from same problem
